apps/e2e/ag-grid-storybook/src/app/ag-grid/ag-grid-stories.component.ts

```diff
@@ -1,79 +1,255 @@
+          },
+          suppressColumnVirtualisation: true,
+          suppressHorizontalScroll: true,
+          suppressRowVirtualisation: true,
+          onGridReady: (params) => {
+            if (dataSet.id === 'row-delete') {
+              params.api.addEventListener(
+                RowNode.EVENT_ROW_SELECTED,
+                ($event: RowSelectedEvent) => {
+                  if ($event.node.isSelected()) {
+                    this.rowDeleteIds = this.rowDeleteIds.concat([
+                      $event.node.id,
+                    ]);
+                  } else {
+                    this.rowDeleteIds = this.rowDeleteIds.filter(
+                      (id) => id !== $event.node.id
+                    );
+                  }
+                }
+              );
+            }
+
+            params.api.addEventListener(
+              Events.EVENT_FIRST_DATA_RENDERED,
+              () => {
+                (
+                  this.#gridsReady.get(dataSet.id) as BehaviorSubject<boolean>
+                ).next(true);
+              }
+            );
+          },
+          rowData: dataSet.data,
+        },
+      });
+    });
   }
 
-  @Input()
-  public domLayout: 'normal' | 'autoHeight' | 'print' = 'autoHeight';
+  public ngAfterViewInit(): void {
+    this.#ngUnsubscribe.add(
+      combineLatest(Array.from(this.#gridsReady.values()))
+        .pipe(
+          filter((gridsReady) => gridsReady.every((ready) => ready)),
+          delay(1000)
+        )
+        .subscribe(() => {
+          // Scroll down to show the back-to-top button.
+          this.#doc
+            .querySelector(
+              '#back-to-top .sky-ag-grid-row-hunteca01 [col-id="name"]'
+            )
+            .scrollIntoView();
 
-  @Input()
-  public enableTopScroll = true;
+          setTimeout(() => {
+            // Select a row to show the row delete button.
+            this.#doc
+              .querySelector(
+                '#row-delete .sky-ag-grid-row-killeha01 [col-id="select"] label'
+              )
+              .dispatchEvent(new MouseEvent('click'));
 
-  @Input()
-  public editable = true;
+            setTimeout(() => {
+              // Trigger validation popover to show up.
+              this.#doc
+                .querySelector(
+                  '#validation [row-index="0"] [col-id="seasons_played"]'
+                )
+                .dispatchEvent(new KeyboardEvent('keyup', { key: 'ArrowUp' }));
 
-  public items = data;
-  public gridOptions: GridOptions;
-  public isActive$ = new BehaviorSubject(true);
-  public ready = new BehaviorSubject(false);
+              // Tell Cypress we're ready.
+              setTimeout(() => this.ready.next(true), 100);
+            });
+          });
+        })
+    );
+    if (!this.skyTheme) {
+      this.isActive$.next(false);
+      this.addPreviewWrapper$.next(true);
+      setTimeout(() => {
+        this.#changeDetectorRef.markForCheck();
+      });
+    }
```

Comment:
Running the SPA as `nx serve ag-grid-storybook` allows for Angular development tools to work, but it does not automatically wrap the page with the theme. This checks if the theme was not loaded and adds the `<sky-preview-wrapper>` to the page.